dsweb/src/main/java/com/doctusoft/dsw/client/gwt/DataTableRenderer.java

```diff
@@ -102,33 +87,47 @@ else if (index == rows.size()) {
 				}
 				rows.add( index, row );
 			}
-			
+
 			@Override
-			public void removed( ObservableList<DataTableRowModel> list, int index, DataTableRowModel element ) {
+			public void removed( final ObservableList<DataTableRowModel> list, final int index, final DataTableRowModel element ) {
 				rows.get( index ).remove();
 				rows.remove( index );
 			}
 		};
 		new ListBindingListener<Integer>( Bindings.obs( model ).get( DataTableModel_._selectedIndices ) ) {
-			
+
 			@Override
-			public void inserted( ObservableList<Integer> list, int index, Integer element ) {
+			public void inserted( final ObservableList<Integer> list, final int index, final Integer element ) {
 				if (rows.size() > element){ //temporary solution, FIXME gabor-farkas https://github.com/Doctusoft/dsweb/issues/20
 					rows.get( element ).addClass( "selected" );
 				}
 			}
-			
+
 			@Override
-			public void removed( ObservableList<Integer> list, int index, Integer element ) {
+			public void removed( final ObservableList<Integer> list, final int index, final Integer element ) {
 				if (rows.size() > element){ //temporary solution, FIXME gabor-farkas https://github.com/Doctusoft/dsweb/issues/20
 					rows.get( element ).removeClass( "selected" );
 				}
 			}
 		};
+
+		new ListBindingListener<DataTableColumnModel>(Bindings.obs(model).get(DataTableModel_._columns)) {
```

Comment:
@mterjeki you can use ListChangeListener for this case so that you don't have to repeat the same code for insert and remove.